libs/vm-modules/tests/unit/math/tensor_tests.cpp

```diff
@@ -468,16 +465,195 @@ TEST_F(MathTensorTests, tensor_sum_etch_test)
     endfunction
   )";
 
-  std::string const state_name{"tensor"};
-
   ASSERT_TRUE(toolkit.Compile(tensor_sum_src));
   Variant res;
   ASSERT_TRUE(toolkit.Run(&res));
 
-  auto const sum_val = res.Get<DataType>();
-  DataType   gt{65.1};
+  auto const     sum_val = res.Get<DataType>();
+  DataType const expected{65.1};
+
+  EXPECT_TRUE(fetch::math::Abs(expected - sum_val) < DataType::TOLERANCE);
+}
+
+TEST_F(MathTensorTests, tensor_transpose_test)
+{
+  fetch::math::Tensor<DataType> tensor =
+      fetch::math::Tensor<DataType>::FromString("1.1, 2.2, 3.3; 4.4, 5.5, 6.6;");
+  ASSERT_TRUE(tensor.shape().size() == 2);
+
+  fetch::vm_modules::math::VMTensor vm_tensor(&toolkit.vm(), 0, tensor);
+
+  auto const transposed = vm_tensor.Transpose()->GetTensor();
+
+  DataType const result = transposed.At<fetch::math::SizeType, fetch::math::SizeType>(1, 0);
+  DataType const expected{2.2};
+
+  EXPECT_EQ(tensor.shape().at(0), transposed.shape().at(1));
+  EXPECT_EQ(tensor.shape().at(1), transposed.shape().at(0));
+
+  EXPECT_TRUE(fetch::math::Abs(expected - result) < DataType::TOLERANCE);
```

Comment:
after konstantinos' parser gets merged in this test can be replaced with EXPECT_TRUE(expected == result)

Reply:
I have replaced this assertion with EXPECT_EQ and it works in all cases but 14273.000000000, so it needs additional investigation.

Reply:
UPD: here is ok to use "==" but still we have cases where delta-based comparison is necessary.